src/go/rpk/pkg/system/metrics.go

```diff
@@ -27,45 +29,123 @@ type Metrics struct {
 	FreeSpaceMB	float64
 }
 
+type stat struct {
+	// Reference: https://man7.org/linux/man-pages/man5/proc.5.html
+
+	// Amount of time that this process has been scheduled
+	// in user mode, measured in clock ticks (divide by
+	// sysconf(_SC_CLK_TCK)).  This includes guest_time (time
+	// spent running a virtual CPU).
+	utime	uint64
+
+	// Amount of time that this process has been scheduled
+	// in kernel mode, measured in clock ticks (divide by
+	// sysconf(_SC_CLK_TCK)).
+	stime	uint64
+}
+
 func GatherMetrics(
 	fs afero.Fs, timeout time.Duration, conf config.Config,
-) (*Metrics, []error) {
+) (*Metrics, error) {
+	var errs error
 	metrics := &Metrics{}
-	errs := []error{}
-	cpuPercentage, err := redpandaCpuPercentage(fs, conf.PIDFile())
+
+	freeDiskSpaceMB, err := getFreeDiskSpaceMB(conf)
```

Comment:
nit: Since the result of the function is not to be used in case of errors, you could directly go with:
`metrics.FreeSpaceMB, err = getFreeDiskSpaceMB(conf)`

Reply:
Great suggestion.